### PR TITLE
🏗Refactor Think Architecture

### PR DESCRIPTION
**Pull request by Vyperus**
**Changes:**

1. Introduce new Think Module.
2. Add new Route. 
3. Delete old Route.
4. Refactor "processdef-list"-module into "diagram-list"-module. 

PR: #PullRequest

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: src/modules/think/think.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import {ISolutionEntry, ISolutionService} from '../../contracts';
+
+export interface IThinkRouteParameters {
+  view?: string;
+  diagramName?: string;
+  solutionUri?: string;
+}
+
+@inject('SolutionService', Router, EventAggregator)
```
Comment (alexanderkasten) [inline src/modules/think/think.ts]:
```suggestion
@inject('SolutionService')
```

File: src/modules/think/diagram-list/diagram-list.ts [hunk 2]
```diff
@@ -1,55 +1,31 @@
...
 
   constructor(eventAggregator: EventAggregator,
-              router: Router,
-              solutionService: ISolutionService) {
+              router: Router) {
     this._eventAggregator = eventAggregator;
     this._router = router;
-    this._solutionService = solutionService;
 
     this._eventAggregator.publish(environment.events.refreshProcessDefs);
```
Comment (alexanderkasten) [inline src/modules/think/diagram-list/diagram-list.ts]: Please remove this event from all used places in BPMN-Studio, there is subscription anymore.
Do not forget the `dev.ts`.

File: src/modules/think/think.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  constructor(solutionService: ISolutionService) {
+    this._solutionService = solutionService;
+  }
+
+  public async activate(routeParameters: IThinkRouteParameters): Promise<void> {
+    const solutionUriIsSet: boolean = routeParameters.solutionUri !== undefined;
+
+    const solutionUri: string = solutionUriIsSet
+                              ? routeParameters.solutionUri
+                              : window.localStorage.getItem('InternalProcessEngineRoute');
```
Comment (cmg-dev) [inline src/modules/think/think.ts]: 👍

File: src/modules/think/think.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  public activeSolutionEntry: ISolutionEntry;
+
+  private _solutionService: ISolutionService;
+
+  constructor(solutionService: ISolutionService) {
+    this._solutionService = solutionService;
+  }
+
+  public async activate(routeParameters: IThinkRouteParameters): Promise<void> {
```
Comment (cmg-dev) [inline src/modules/think/think.ts]: I like the interface `IThinkRouteParameters`. This is a clever usage of TS-Interfaces.

**Pull request merged**